quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/integration/jaxrs/QuarkusKeycloakApplication.java

```diff
@@ -69,13 +66,6 @@ public Set<Object> getSingletons() {
 
     @Override
     public Set<Class<?>> getClasses() {
-        Set<Class<?>> classes = new HashSet<>(super.getClasses());
-
-        classes.add(QuarkusObjectMapperResolver.class);
-        classes.add(CloseSessionHandler.class);
-
-        classes.add(DebugHostnameSettingsResource.class);
-
-        return classes;
+        return Set.of();
```

Comment:
This approach seems clear and elegant but I can see one major drawback to it: duplicated logic for provider classes. `KeycloakApplication` uses `getClasses` while `QuarkusKeycloakApplication` uses annotations and filters out the classes in `KeycloakProcessor`. AFAIK `KeycloakApplication` serves now just for the Undertow auth server, i.e. just for dev purposes. I wonder if Undertow can't use at least the jax-rs annotations as well.

Reply:
The issue is with the conditional inclusion of resources. See #27242 - we'll be able at that point to remove the classes / singletons from KeycloakApplication entirely.